hpccm/building_blocks/scif.py

```diff
@@ -0,0 +1,122 @@
+# pylint: disable=invalid-name, too-few-public-methods
+# pylint: disable=too-many-instance-attributes
+
+"""SCIF building block"""
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from __future__ import print_function
+
+from hpccm.building_blocks.packages import packages
+from hpccm.common import container_type
+from hpccm.primitives.comment import comment
+from hpccm.primitives.copy import copy
+from hpccm.primitives.runscript import runscript
+from hpccm.primitives.shell import shell
+
+import hpccm.config
+import logging
+import os
+
+class scif(object):
+    """SCIF building block"""
+    initialized = False
+
+    def __init__(self, **kwargs):
+        """Initialize building block"""
+        self.name = kwargs.get('name', '')
+        self.scif_version = kwargs.get('scif_version', '0.0.76')
+        self.__layers = []
+        self.__separator = kwargs.get('separator', '\n\n')
+
+        self.__entrypoint = kwargs.get('entrypoint', False)
+        self.instructions = [comment("SCIF app {0}".format(self.name))]
+        if hpccm.config.g_ctype == container_type.DOCKER and not scif.initialized:
+            # Install the scif-tool on first scif app
+            self.instructions.extend([
+                comment("Begin SCI-F installation"),
+                packages(ospackages=['python-pip', 'python-setuptools']),
+                shell(commands=[
+                  'pip install wheel',
+                  'pip install scif=={0}'.format(self.scif_version)
+                ]),
+                comment("End SCI-F installation")
```

Comment:
FYI: A pip building block was recently added.  Not sure if it would be appropriate to use here or not, so just making sure you're aware.

One approach would be to remove the pip stuff from the scif building block and just say in the documentation that you need to call `pip(packages=['scif'])` before using the scif building block.  There is precedence for this with the MPI and OFED building blocks or some of the IO building blocks and HDF5, for example.

On the other hand, it is convenient to have scif installed automatically as part of the scif building block.   It's your call.